### PR TITLE
Adds an option to override the real precision to r4

### PR DESCRIPTION
**Pull request by uramirez8707**
This can be used in an xmls by doing 
```
<makeOverrides>USE_R4="ON"</makeOverrides>
```

For the component that you want to use r4 for. 

This option will allow us to compile different components at different precisions (i.e compile fv3 in r4 and everything else in r8)

### CODE REVIEW

**Comment by uramirez8707** [comment]
@bensonr @underwoo

**Comment by uramirez8707** [comment]
> While this looks good for Intel on c5, what about the other templates?

@bensonr Should all templates be updated?

File: templates/ncrc5-intel-classic.mk [hunk 1]
```diff
@@ -77,6 +79,14 @@ $(error Options DEBUG and TEST cannot be used together)
...
 endif
 
+
+ifdef USE_R4
+REAL_PRECISION := -r4
+CPPDEFS += -DOVERLOAD_R4
+else
+REAL_PRECISION := -r8
+endif
+
```
Comment (uramirez8707) [inline templates/ncrc5-intel-classic.mk:89]: @bensonr 
should we be using `-real-size 32` and `real-size 64` instead?

Comment (bensonr) [reply templates/ncrc5-intel-classic.mk:89]: @uramirez8707 - Yes, I think we should be using the updated `-real-size` for both OneAPI and classic ifort.

**Review by bensonr — APPROVED** [review]

**Pull request merged**